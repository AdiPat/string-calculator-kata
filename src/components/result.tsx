export function Result({ result }: { result: number }) {
  return (
    <div className="text-gray-800 text-center mt-4 bg-gray-400 rounded-lg p-4">
      <p className="text-xl font-bold">Result: {result}</p>
    </div>
  );
}
