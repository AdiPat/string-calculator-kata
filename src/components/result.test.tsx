import { Result } from "./result";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("Result Tests", () => {
  it("renders a result with a number", async () => {
    const { container } = render(<Result result={0} />);
    expect(container).toMatchSnapshot();
  });

  it("renders a result with a number", async () => {
    const { container } = render(<Result result={0} />);
    const result = screen.getByText(/Result: 0/i);
    expect(result).toBeInTheDocument();
  });
});
