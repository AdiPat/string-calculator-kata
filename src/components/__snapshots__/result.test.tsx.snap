// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Tests renders a result with a number 1`] = `
<div>
  <div
    class="text-gray-800 text-center mt-4 bg-gray-400 rounded-lg p-4"
  >
    <p
      class="text-xl font-bold"
    >
      Result: 
      0
    </p>
  </div>
</div>
`;
